esiclient/v1/node_baremetal.py

```diff
@@ -0,0 +1,73 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import logging
+
+import openstack
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+# Initialize and turn on debug logging
+openstack.enable_logging(debug=False)
+
+
+class MDCBaremetalNodeList(command.Lister):
+    """List baremetal nodes from multiple OpenStack instances"""
+
+    log = logging.getLogger(__name__ + ".List")
+
+    def get_parser(self, prog_name):
+        parser = super(MDCBaremetalNodeList, self).get_parser(prog_name)
+        parser.add_argument(
+            '--clouds',
+            dest='clouds',
+            metavar='<clouds>',
+            nargs="+",
+            required=True,
+            help=_("Specify the cloud to use from clouds.yaml.")
+        )
+
+        parser.add_argument(
+            '--regions',
```

Comment:
it looks like region is something that's part of clouds.yaml - can it be taken from there?

Reply:
I'd say that the user should have multiple clouds.yaml entries then, one for each region. Regardless if this is an option or not, we want the default to be as simple as possible for the end user, and having them list a region for each cloud they want to use isn't all that simple.

Reply:
It seems to be part of a normal workflow to have one cloud entry with multiple regions listed, the documentation for using the openstack.connect library gives an example using such a setup

Reply:
In that case we should iterate over the regionName; if that's not possible then we should document that the CLI doesn't support multiple regions in a cloud. Having to list all the region names for multiple clouds each time a user runs the CLI isn't user-friendly, and that's the primary goal of this feature.